// calcite-react/Modal/Modal-styled.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DialogTransition = exports.OverlayTransition = exports.StyledModalActions = exports.StyledModalDialog = exports.StyledModalOverlay = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _CalciteThemeProvider = require("../CalciteThemeProvider");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var StyledModalOverlay = {
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
  position: 'fixed',
  top: '0px',
  left: '0px',
  right: '0px',
  bottom: '0px',
  textAlign: 'center',
  background: _CalciteThemeProvider.CalciteTheme.palette.transparentBlack,
  zIndex: 101,
  transition: "opacity 300ms cubic-bezier(0.4, 0.0, 0.2, 1)",
  opacity: 0
};
exports.StyledModalOverlay = StyledModalOverlay;
var StyledModalDialog = {
  position: 'relative',
  top: 'auto',
  left: 'auto',
  right: 'auto',
  bottom: 'auto',
  maxWidth: '80vw',
  maxHeight: '80vh',
  overflow: 'auto',
  outline: 'none',
  boxSizing: 'border-box',
  zIndex: '102',
  background: _CalciteThemeProvider.CalciteTheme.palette.white,
  padding: _CalciteThemeProvider.CalciteTheme.baseline,
  textAlign: 'left',
  overflowY: 'auto',
  display: 'inline-block',
  verticalAlign: 'middle',
  borderRadius: _CalciteThemeProvider.CalciteTheme.borderRadius,
  border: 'none',
  transition: "margin-top 300ms cubic-bezier(0.4, 0.0, 0.2, 1)",
  marginTop: '30px'
};
exports.StyledModalDialog = StyledModalDialog;

var StyledModalActions = _styledComponents.default.div.withConfig({
  displayName: "StyledModalActions",
  componentId: "sc-1qolgd0-0"
})(["display:flex;justify-content:flex-end;align-items:center;"]);

exports.StyledModalActions = StyledModalActions;
var OverlayTransition = {
  entering: {
    opacity: 0
  },
  entered: {
    opacity: 1
  },
  exiting: {
    opacity: 0
  }
};
exports.OverlayTransition = OverlayTransition;
var DialogTransition = {
  entering: {
    marginTop: '30px'
  },
  entered: {
    marginTop: 0
  },
  exiting: {
    marginTop: '30px'
  }
};
exports.DialogTransition = DialogTransition;
