// calcite-react/CopyToClipboard/CopyToClipboard-styled.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StyledCopyButton = exports.ClipboardIconStyles = exports.StyledCopyToClipboardInput = exports.StyledCopyToClipboard = void 0;

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _commonElements = require("../utils/commonElements");

var _helpers = require("../utils/helpers");

var _Button = _interopRequireDefault(require("../Button"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

var FadeIn = (0, _styledComponents.keyframes)(["from{opacity:0;}to{opacity:1;}"]);

var StyledCopyToClipboard = _styledComponents.default.div.withConfig({
  displayName: "StyledCopyToClipboard",
  componentId: "sc-1nvred8-0"
})(["display:flex;svg{animation:", " ", ";}"], FadeIn, (0, _helpers.transition)());

exports.StyledCopyToClipboard = StyledCopyToClipboard;
var StyledCopyToClipboardInput = (0, _styledComponents.default)(_commonElements.CalciteInput).withConfig({
  displayName: "StyledCopyToClipboardInput",
  componentId: "sc-1nvred8-1"
})(["flex:1 0 50px;width:auto;margin:0;border-right:none;border-top-right-radius:0;border-bottom-right-radius:0;"]);
exports.StyledCopyToClipboardInput = StyledCopyToClipboardInput;
var StyledCopyButton = (0, _styledComponents.default)(_Button.default).withConfig({
  displayName: "StyledCopyButton",
  componentId: "sc-1nvred8-2"
})(["border-top-left-radius:0;border-bottom-left-radius:0;"]);
exports.StyledCopyButton = StyledCopyButton;
var ClipboardIconStyles = {
  marginLeft: 0
};
exports.ClipboardIconStyles = ClipboardIconStyles;
