// react-accessible-tabs/lib/components/Panels.js
'use strict';

exports.__esModule = true;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _reactAddonsCreateFragment = require('react-addons-create-fragment');

var _reactAddonsCreateFragment2 = _interopRequireDefault(_reactAddonsCreateFragment);

var _Panel = require('./Panel');

var _Panel2 = _interopRequireDefault(_Panel);

var _idSafeName = require('../helpers/idSafeName');

var _idSafeName2 = _interopRequireDefault(_idSafeName);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Panels = function (_Component) {
    _inherits(Panels, _Component);

    function Panels() {
        _classCallCheck(this, Panels);

        return _possibleConstructorReturn(this, _Component.apply(this, arguments));
    }

    Panels.prototype.render = function render() {
        var _props = this.props;
        var data = _props.data;
        var selectedIndex = _props.selectedIndex;


        if (!data.length) {
            return null;
        }

        return _react2.default.createElement(
            'div',
            { className: 'tabs__panels' },
            data.map(function (panel, index) {
                var id = (0, _idSafeName2.default)(panel.label, index);
                var o = {};

                o[id] = panel.content;
                var children = (0, _reactAddonsCreateFragment2.default)(o);

                return _react2.default.createElement(
                    _Panel2.default,
                    {
                        key: id,
                        id: id,
                        index: index,
                        selectedIndex: selectedIndex },
                    children
                );
            })
        );
    };

    return Panels;
}(_react.Component);

process.env.NODE_ENV !== "production" ? Panels.propTypes = {
    data: _propTypes2.default.array,
    selectedIndex: _propTypes2.default.number
} : void 0;

exports.default = Panels;
module.exports = exports['default'];
