// react-accessible-tabs/lib/components/Panel.js
'use strict';

exports.__esModule = true;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Panel = function (_Component) {
    _inherits(Panel, _Component);

    function Panel() {
        _classCallCheck(this, Panel);

        return _possibleConstructorReturn(this, _Component.apply(this, arguments));
    }

    Panel.prototype.createMarkup = function createMarkup(str) {
        return { __html: str };
    };

    Panel.prototype.renderContents = function renderContents(child) {
        return _react2.default.isValidElement(child) ? child : _react2.default.createElement('div', { dangerouslySetInnerHTML: this.createMarkup(child) });
    };

    Panel.prototype.render = function render() {
        var _props = this.props;
        var children = _props.children;
        var id = _props.id;
        var index = _props.index;
        var selectedIndex = _props.selectedIndex;

        var isSelected = index === selectedIndex;
        var className = (0, _classnames2.default)('tabs__panel', { 'is-hidden': !isSelected });

        return _react2.default.createElement(
            'section',
            {
                'aria-hidden': !isSelected,
                className: className,
                id: id,
                role: 'tabpanel',
                tabIndex: isSelected ? 0 : -1 },
            _react2.default.Children.map(children, this.renderContents, this)
        );
    };

    return Panel;
}(_react.Component);

process.env.NODE_ENV !== "production" ? Panel.propTypes = {
    children: _propTypes2.default.oneOfType([_propTypes2.default.element, _propTypes2.default.string, _propTypes2.default.arrayOf(_propTypes2.default.oneOfType([_propTypes2.default.element, _propTypes2.default.string]))]),
    id: _propTypes2.default.string,
    index: _propTypes2.default.number,
    selectedIndex: _propTypes2.default.number
} : void 0;

exports.default = Panel;
module.exports = exports['default'];
