// react-accessible-tabs/lib/components/Tabs.js
'use strict';

exports.__esModule = true;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _TabList = require('./TabList');

var _TabList2 = _interopRequireDefault(_TabList);

var _Panels = require('./Panels');

var _Panels2 = _interopRequireDefault(_Panels);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Tabs = function (_Component) {
    _inherits(Tabs, _Component);

    function Tabs(props) {
        _classCallCheck(this, Tabs);

        var _this = _possibleConstructorReturn(this, _Component.call(this, props));

        _this.handleClick = function (e, index) {
            e.preventDefault();

            if (_this.state.selectedIndex === index) {
                return;
            }

            _this.setState({
                userInvokedSelection: true,
                selectedIndex: index
            });
        };

        _this.handleKeyDown = function (e) {
            if (e.key === 'ArrowLeft' || e.key === 'ArrowRight') {
                e.preventDefault();
            } else {
                return;
            }

            var targetIndex = void 0;

            if (e.key === 'ArrowLeft' && _this.state.selectedIndex > 0) {
                targetIndex = _this.state.selectedIndex - 1;
            } else if (e.key === 'ArrowRight' && _this.state.selectedIndex < _this.tabTriggersLength - 1) {
                targetIndex = _this.state.selectedIndex + 1;
            } else {
                return;
            }

            _this.setState({
                userInvokedSelection: true,
                selectedIndex: targetIndex
            });
        };

        _this.resetUserInvokedSelection = function () {
            _this.setState({
                userInvokedSelection: false
            });
        };

        _this.tabTriggersLength = _this.props.data.length;

        _this.state = {
            userInvokedSelection: false,
            selectedIndex: _this.props.initialSelectedIndex < 0 || _this.props.initialSelectedIndex > _this.tabTriggersLength - 1 ? 0 : _this.props.initialSelectedIndex
        };
        return _this;
    }

    Tabs.prototype.render = function render() {
        if (!this.tabTriggersLength) {
            return null;
        }

        return _react2.default.createElement(
            'div',
            { className: 'tabs' },
            _react2.default.createElement(_TabList2.default, _extends({}, this.props, {
                userInvokedSelection: this.state.userInvokedSelection,
                selectedIndex: this.state.selectedIndex,
                onClick: this.handleClick,
                onKeyDown: this.handleKeyDown,
                resetUserInvokedSelection: this.resetUserInvokedSelection
            })),
            _react2.default.createElement(_Panels2.default, _extends({}, this.props, { selectedIndex: this.state.selectedIndex }))
        );
    };

    return Tabs;
}(_react.Component);

Tabs.defaultProps = {
    data: [],
    initialSelectedIndex: 0
};

process.env.NODE_ENV !== "production" ? Tabs.propTypes = {
    data: _propTypes2.default.arrayOf(_propTypes2.default.shape({
        label: _propTypes2.default.string,
        content: _propTypes2.default.oneOfType([_propTypes2.default.element, _propTypes2.default.string, _propTypes2.default.arrayOf(_propTypes2.default.oneOfType([_propTypes2.default.element, _propTypes2.default.string]))])
    })),
    selectedIndex: _propTypes2.default.number,
    initialSelectedIndex: _propTypes2.default.number
} : void 0;

exports.default = Tabs;
module.exports = exports['default'];
