// calcite-react/Loader/Loader-styled.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StyledLoaderText = exports.StyledLoaderBars = exports.StyledLoader = void 0;

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _helpers = require("../utils/helpers");

var _CalciteThemeProvider = require("../CalciteThemeProvider");

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

var loaderVariables = {
  loaderWidth: '0.85rem',
  loaderHeight: '2rem',
  loaderZoom: '0.5rem',
  loaderSpacing: '1.25rem',
  loaderSpeed: '0.8s',
  loaderDelay: '0.16s',
  loaderBlue: _CalciteThemeProvider.CalciteTheme.palette.blue
};
var loadKeyframes = (0, _styledComponents.keyframes)(["0%,80%,100%{opacity:.75;box-shadow:0 0 ", ";height:", ";}40%{opacity:1;box-shadow:0 -0.5rem ", ";height:", ";}"], loaderVariables.loaderBlue, loaderVariables.loaderHeight, loaderVariables.loaderBlue, (0, _helpers.unitCalc)(loaderVariables.loaderHeight, loaderVariables.loaderZoom, '+'));
var loadAnimation = (0, _styledComponents.css)(["", " ", " infinite ease-in-out;"], loadKeyframes, loaderVariables.loaderSpeed);
var loaderStyles = (0, _styledComponents.css)(["background:", ";animation:", ";width:", ";height:", ";"], loaderVariables.loaderBlue, loadAnimation, loaderVariables.loaderWidth, loaderVariables.loaderHeight);
var loaderPseudoElements = (0, _styledComponents.css)(["", ";position:absolute;top:0;content:'';"], loaderStyles);

var StyledLoader = _styledComponents.default.div.withConfig({
  displayName: "StyledLoader",
  componentId: "v05kzr-0"
})(["position:relative;min-height:3rem;"]);

exports.StyledLoader = StyledLoader;

var StyledLoaderBars = _styledComponents.default.div.withConfig({
  displayName: "StyledLoaderBars",
  componentId: "v05kzr-1"
})(["", ";text-indent:-9999em;margin:auto;position:absolute;right:", ";font-size:11px;animation-delay:", ";margin:", " 0 0;&:before{", ";left:-", ";}&:after{", ";left:", ";animation-delay:", ";}"], loaderStyles, function (props) {
  return "calc(50% - ".concat(parseFloat(loaderVariables.loaderWidth) / 2, "rem)");
}, loaderVariables.loaderDelay, function (props) {
  return (0, _helpers.unitCalc)(loaderVariables.loaderHeight, 0.25, '*');
}, loaderPseudoElements, loaderVariables.loaderSpacing, loaderPseudoElements, loaderVariables.loaderSpacing, (0, _helpers.unitCalc)(loaderVariables.loaderDelay, 2, '*'));

exports.StyledLoaderBars = StyledLoaderBars;

var StyledLoaderText = _styledComponents.default.div.withConfig({
  displayName: "StyledLoaderText",
  componentId: "v05kzr-2"
})(["text-align:center;padding-top:3.5rem;"]);

exports.StyledLoaderText = StyledLoaderText;
