// calcite-react/Select/Select-styled.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PopperManagerStyles = exports.StyledSelectMenu = exports.StyledSelectButton = exports.StyledSelectInput = exports.StyledSelectWrapper = void 0;

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _commonElements = require("../utils/commonElements");

var _Menu = _interopRequireDefault(require("../Menu"));

var _helpers = require("../utils/helpers");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

var StyledSelectWrapper = _styledComponents.default.div.withConfig({
  displayName: "StyledSelectWrapper",
  componentId: "sc-1mn2ays-0"
})(["position:relative;", ";"], function (props) {
  return props.fullWidth && (0, _styledComponents.css)(["width:100%;"]);
});

exports.StyledSelectWrapper = StyledSelectWrapper;
var StyledSelectInput = (0, _styledComponents.default)(_commonElements.CalciteSelect).withConfig({
  displayName: "StyledSelectInput",
  componentId: "sc-1mn2ays-1"
})(["cursor:pointer;text-overflow:ellipsis;"]);
exports.StyledSelectInput = StyledSelectInput;
var StyledSelectButton = (0, _styledComponents.default)(_commonElements.CalciteSelect).withConfig({
  displayName: "StyledSelectButton",
  componentId: "sc-1mn2ays-2"
})(["cursor:pointer;text-align:left;overflow:hidden;white-space:nowrap;text-overflow:ellipsis;"]);
exports.StyledSelectButton = StyledSelectButton;
var StyledSelectMenu = (0, _styledComponents.default)(_Menu.default).withConfig({
  displayName: "StyledSelectMenu",
  componentId: "sc-1mn2ays-3"
})(["max-height:300px;display:none;border-bottom:none;box-shadow:", ",0 1px 0 ", ";transition:opacity ", ";z-index:2000;", ";", ";"], function (props) {
  return props.theme.boxShadow;
}, function (props) {
  return props.theme.palette.lightestGray;
}, (0, _helpers.transition)(), function (props) {
  return props.isOpen && (0, _styledComponents.css)(["display:block;"]);
}, function (props) {
  return props.fullWidth && (0, _styledComponents.css)(["min-width:100%;"]);
});
exports.StyledSelectMenu = StyledSelectMenu;
var PopperManagerStyles = {
  width: '100%'
};
exports.PopperManagerStyles = PopperManagerStyles;
