// calcite-react/List/List-styled.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StyledListSideContainer = exports.StyledListTextContainer = exports.StyledListSubtitle = exports.StyledListTitle = exports.StyledListHeader = exports.StyledListItem = exports.StyledList = void 0;

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _SideNavStyled = require("../SideNav/SideNav-styled");

var _helpers = require("../utils/helpers");

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

var StyledList = (0, _styledComponents.default)(_SideNavStyled.StyledSideNav).withConfig({
  displayName: "StyledList",
  componentId: "sc-1sb8owe-0"
})(["min-width:200px;overflow:hidden;max-height:", ";transition:max-height 225ms cubic-bezier(0.4,0,0.2,1);", ";"], function (props) {
  return props.maxHeight;
}, function (props) {
  return props.nested && (0, _styledComponents.css)(["border:none;border-top:1px solid ", ";border-top-left-radius:0;border-top-right-radius:0;padding-left:", ";", ";"], function (props) {
    return props.theme.palette.lightestGray;
  }, function (props) {
    return props.theme.baseline;
  }, function (props) {
    return props.open === false && (0, _styledComponents.css)(["border-top:none;"]);
  });
});
exports.StyledList = StyledList;

var getActiveStyles = function getActiveStyles(props) {
  return "\n    background-color: ".concat(props.theme.palette.offWhite, ";\n    text-decoration: none;\n    color: ").concat(props.theme.palette.darkerGray, ";\n  ");
};

var StyledListItem = _styledComponents.default.div.withConfig({
  displayName: "StyledListItem",
  componentId: "sc-1sb8owe-1"
})(["display:flex;justify-content:space-between;box-sizing:border-box;padding:", " ", ";color:", ";background-color:", ";border-top:1px solid ", ";cursor:pointer;", ";&:hover{", ";}", ";", ";"], function (props) {
  return (0, _helpers.unitCalc)(props.theme.baseline, 3, '/');
}, function (props) {
  return (0, _helpers.unitCalc)(props.theme.baseline, 2, '/');
}, function (props) {
  return props.theme.palette.darkerGray;
}, function (props) {
  return props.theme.palette.white;
}, function (props) {
  return props.theme.palette.lightestGray;
}, function (props) {
  return props.active && (0, _styledComponents.css)(["", ";"], function (props) {
    return getActiveStyles(props);
  });
}, function (props) {
  return getActiveStyles(props);
}, function (props) {
  return props.selected && (0, _styledComponents.css)(["&,&:focus,&:hover{text-indent:-3px;border-left:3px solid ", ";}"], function (props) {
    return props.theme.palette.blue;
  });
}, function (props) {
  return props.nested && (0, _styledComponents.css)(["&:first-child{border-top:none;}"]);
});

exports.StyledListItem = StyledListItem;
var StyledListHeader = (0, _styledComponents.default)(StyledListItem).withConfig({
  displayName: "StyledListHeader",
  componentId: "sc-1sb8owe-2"
})(["background-color:", ";cursor:auto;"], function (props) {
  return props.theme.palette.offWhite;
});
exports.StyledListHeader = StyledListHeader;

var StyledListTitle = _styledComponents.default.p.withConfig({
  displayName: "StyledListTitle",
  componentId: "sc-1sb8owe-3"
})(["", ";margin:0;"], (0, _helpers.fontSize)(-1));

exports.StyledListTitle = StyledListTitle;

var StyledListSubtitle = _styledComponents.default.span.withConfig({
  displayName: "StyledListSubtitle",
  componentId: "sc-1sb8owe-4"
})(["", ";line-height:1.2rem;color:", ";"], (0, _helpers.fontSize)(-3), function (props) {
  return props.theme.palette.gray;
});

exports.StyledListSubtitle = StyledListSubtitle;

var StyledListTextContainer = _styledComponents.default.div.withConfig({
  displayName: "StyledListTextContainer",
  componentId: "sc-1sb8owe-5"
})(["display:flex;flex:1 0 100px;flex-direction:column;justify-content:center;"]);

exports.StyledListTextContainer = StyledListTextContainer;

var StyledListSideContainer = _styledComponents.default.div.withConfig({
  displayName: "StyledListSideContainer",
  componentId: "sc-1sb8owe-6"
})(["display:flex;align-items:center;justify-content:center;&:first-child{margin-right:", ";}&:last-child{margin-left:", ";}.mdi-icon{width:28px;height:28px;fill:currentColor;", ";}"], function (props) {
  return (0, _helpers.unitCalc)(props.theme.baseline, 2, '/');
}, function (props) {
  return (0, _helpers.unitCalc)(props.theme.baseline, 2, '/');
}, function (props) {
  return props.nested && (0, _styledComponents.css)(["width:22px;height:22px;"]);
});

exports.StyledListSideContainer = StyledListSideContainer;
