// calcite-react/Search/Search-styled.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PopperStyle = exports.ManagerStyle = exports.StyledMagnifyIcon = exports.StyledCloseCircleIcon = exports.StyledShortcutCharacter = exports.StyledSearch = exports.StyledSearchInputWrapper = exports.StyledSearchContainer = void 0;

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _commonElements = require("../utils/commonElements");

var _helpers = require("../utils/helpers");

var _XCircleIcon = _interopRequireDefault(require("calcite-ui-icons-react/XCircleIcon"));

var _MagnifyingGlassIcon = _interopRequireDefault(require("calcite-ui-icons-react/MagnifyingGlassIcon"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

var StyledCloseCircleIcon = (0, _styledComponents.default)(_XCircleIcon.default).withConfig({
  displayName: "StyledCloseCircleIcon",
  componentId: "sc-1pd0iuk-0"
})(["display:none;position:absolute;right:", ";bottom:0.55em;color:", ";cursor:pointer;&:hover{color:", ";}"], function (props) {
  return (0, _helpers.unitCalc)(props.theme.baseline, 4, '/');
}, function (props) {
  return props.theme.palette.lighterGray;
}, function (props) {
  return props.theme.palette.gray;
});
exports.StyledCloseCircleIcon = StyledCloseCircleIcon;
var StyledMagnifyIcon = (0, _styledComponents.default)(_MagnifyingGlassIcon.default).withConfig({
  displayName: "StyledMagnifyIcon",
  componentId: "sc-1pd0iuk-1"
})(["position:absolute;bottom:0.65em;left:0.5em;color:", ";"], function (props) {
  return props.theme.palette.darkerGray;
});
exports.StyledMagnifyIcon = StyledMagnifyIcon;

var StyledShortcutCharacter = _styledComponents.default.div.withConfig({
  displayName: "StyledShortcutCharacter",
  componentId: "sc-1pd0iuk-2"
})(["position:absolute;right:", ";bottom:0.5em;border:1px solid ", ";display:flex;align-items:center;justify-content:center;width:18px;height:18px;color:", ";border-radius:1px;"], function (props) {
  return (0, _helpers.unitCalc)(props.theme.baseline, 4, '/');
}, function (props) {
  return props.theme.palette.lighterGray;
}, function (props) {
  return props.theme.palette.lightGray;
});

exports.StyledShortcutCharacter = StyledShortcutCharacter;

var StyledSearchContainer = _styledComponents.default.div.withConfig({
  displayName: "StyledSearchContainer",
  componentId: "sc-1pd0iuk-3"
})(["display:inline-flex;align-items:center;position:relative;", ";", "{", ";}", "{", ";}&:hover{", "{display:block;}}&:active,&:focus,&:focus-within{", "{display:none;}}"], function (props) {
  return props.fullWidth && (0, _styledComponents.css)(["width:100%;"]);
}, StyledMagnifyIcon, function (props) {
  return props.minimal && (0, _styledComponents.css)(["bottom:0.75em;left:0.1rem;"]);
}, StyledCloseCircleIcon, function (props) {
  return props.minimal && (0, _styledComponents.css)(["bottom:auto;"]);
}, StyledCloseCircleIcon, StyledShortcutCharacter);

exports.StyledSearchContainer = StyledSearchContainer;

var StyledSearchInputWrapper = _styledComponents.default.div.withConfig({
  displayName: "StyledSearchInputWrapper",
  componentId: "sc-1pd0iuk-4"
})(["position:relative;flex:1 0 50px;"]);

exports.StyledSearchInputWrapper = StyledSearchInputWrapper;
var StyledSearch = (0, _styledComponents.default)(_commonElements.CalciteInput).withConfig({
  displayName: "StyledSearch",
  componentId: "sc-1pd0iuk-5"
})(["padding-left:", ";padding-right:", ";background:transparent;", ";"], function (props) {
  return (0, _helpers.unitCalc)(props.theme.baseline, 1.2, '*');
}, function (props) {
  return props.theme.baseline;
}, function (props) {
  return props.minimal && (0, _styledComponents.css)(["height:2.35rem;border-bottom:2px solid ", ";", ";&:focus{box-shadow:none;border-bottom-color:", ";}"], function (props) {
    return props.theme.palette.lighterGray;
  }, (0, _helpers.fontSize)(0), function (props) {
    return props.theme.palette.blue;
  });
});
exports.StyledSearch = StyledSearch;
var ManagerStyle = {
  width: '100%'
};
exports.ManagerStyle = ManagerStyle;
var PopperStyle = {
  transition: "opacity ".concat((0, _helpers.transition)()),
  zIndex: 2000
};
exports.PopperStyle = PopperStyle;
