// calcite-react/ArcgisShare/ArcgisShare-styled.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StyledLegend = exports.StyledStarIcon = exports.GroupFieldsetStyles = exports.GroupCheckboxLabelStyles = exports.PrimaryCheckboxLabelStyles = exports.StyledGroupContainer = exports.StyledArcgisShare = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _CalciteThemeProvider = require("../CalciteThemeProvider");

var _helpers = require("../utils/helpers");

var _StarIcon = _interopRequireDefault(require("calcite-ui-icons-react/StarIcon"));

var _Form = require("../Form");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var StyledArcgisShare = _styledComponents.default.div.withConfig({
  displayName: "StyledArcgisShare",
  componentId: "ymkgyh-0"
})([""]);

exports.StyledArcgisShare = StyledArcgisShare;

var StyledGroupContainer = _styledComponents.default.div.withConfig({
  displayName: "StyledGroupContainer",
  componentId: "ymkgyh-1"
})(["border:1px solid ", ";padding:", ";min-width:180px;max-height:150px;min-height:95px;overflow-y:auto;border-radius:", ";"], function (props) {
  return props.theme.palette.lightestGray;
}, function (props) {
  return (0, _helpers.unitCalc)(props.theme.baseline, 6, '/');
}, function (props) {
  return props.theme.borderRadius;
});

exports.StyledGroupContainer = StyledGroupContainer;
var PrimaryCheckboxLabelStyles = {
  fontSize: '1rem',
  color: _CalciteThemeProvider.CalciteTheme.palette.black
};
exports.PrimaryCheckboxLabelStyles = PrimaryCheckboxLabelStyles;
var GroupCheckboxLabelStyles = {
  fontSize: '0.85rem'
};
exports.GroupCheckboxLabelStyles = GroupCheckboxLabelStyles;
var GroupFieldsetStyles = {
  marginLeft: '1.4rem',
  display: 'flex',
  alignItems: 'flex-start'
};
exports.GroupFieldsetStyles = GroupFieldsetStyles;
var StyledStarIcon = (0, _styledComponents.default)(_StarIcon.default).withConfig({
  displayName: "StyledStarIcon",
  componentId: "ymkgyh-2"
})(["fill:", ";vertical-align:text-top;margin-left:2px;"], function (props) {
  return props.theme.palette.lighterGray;
});
exports.StyledStarIcon = StyledStarIcon;
var StyledLegend = (0, _styledComponents.default)(_Form.Legend).withConfig({
  displayName: "StyledLegend",
  componentId: "ymkgyh-3"
})(["font-size:1rem;"]);
exports.StyledLegend = StyledLegend;
